modules/dxp/apps/segments/segments-asah-connector-test/src/testIntegration/java/com/liferay/segments/asah/connector/internal/model/listener/test/SegmentsExperimentModelListenerTest.java

```diff
@@ -0,0 +1,280 @@
+			asahSegmentsExperimentProcessor, "_asahFaroBackendClient");
+
+		ReflectionTestUtil.setFieldValue(
+			asahFaroBackendClient, "_http",
+			MockHttpUtil.geHttp(
+				Collections.singletonMap(
+					"/api/1.0/experiments/" +
+						segmentsExperiment1.getSegmentsExperimentKey(),
+					() -> "{}")));
+
+		try (CompanyConfigurationTemporarySwapper
+				companyConfigurationTemporarySwapper =
+					new CompanyConfigurationTemporarySwapper(
+						TestPropsValues.getCompanyId(),
+						AnalyticsConfiguration.class.getName(),
+						HashMapDictionaryBuilder.<String, Object>put(
+							"liferayAnalyticsDataSourceId",
+							RandomTestUtil.randomLong()
+						).put(
+							"liferayAnalyticsEnableAllGroupIds", true
+						).put(
+							"liferayAnalyticsFaroBackendSecuritySignature",
+							RandomTestUtil.randomString()
+						).put(
+							"liferayAnalyticsFaroBackendURL",
+							RandomTestUtil.randomString()
+						).build(),
+						SettingsFactoryUtil.getSettingsFactory())) {
+
+			String description = "A/B test description updated";
+
+			segmentsExperiment1.setDescription(description);
+
+			_segmentsExperimentLocalService.updateSegmentsExperiment(
+				segmentsExperiment1);
+
+			SegmentsExperiment segmentsExperiment2 =
+				_segmentsExperimentLocalService.getSegmentsExperiment(
+					segmentsExperiment1.getSegmentsExperimentId());
+
+			Assert.assertEquals(
+				description, segmentsExperiment2.getDescription());
+		}
+		finally {
+			ReflectionTestUtil.setFieldValue(
+				asahFaroBackendClient, "_http", _http);
+		}
+	}
+
+	@Test
+	public void testOnBeforeCreate() throws Exception {
+		Object asahSegmentsExperimentProcessor =
+			ReflectionTestUtil.getFieldValue(
+				_modelListener, "_asahSegmentsExperimentProcessor");
+
+		Object asahFaroBackendClient = ReflectionTestUtil.getFieldValue(
+			asahSegmentsExperimentProcessor, "_asahFaroBackendClient");
+
+		String segmentsExperimentKey = "123456789";
+
+		ReflectionTestUtil.setFieldValue(
+			asahFaroBackendClient, "_http",
+			MockHttpUtil.geHttp(
+				Collections.singletonMap(
+					"/api/1.0/experiments",
+					() -> JSONUtil.put(
+						"id", segmentsExperimentKey
+					).toString())));
+
+		try (CompanyConfigurationTemporarySwapper
+				companyConfigurationTemporarySwapper =
+					new CompanyConfigurationTemporarySwapper(
+						TestPropsValues.getCompanyId(),
+						AnalyticsConfiguration.class.getName(),
+						HashMapDictionaryBuilder.<String, Object>put(
+							"liferayAnalyticsDataSourceId",
+							RandomTestUtil.randomLong()
+						).put(
+							"liferayAnalyticsEnableAllGroupIds", true
+						).put(
+							"liferayAnalyticsFaroBackendSecuritySignature",
+							RandomTestUtil.randomString()
+						).put(
+							"liferayAnalyticsFaroBackendURL",
+							RandomTestUtil.randomString()
+						).build(),
+						SettingsFactoryUtil.getSettingsFactory())) {
+
+			SegmentsExperiment segmentsExperiment = _addSegmentsExperiment();
+
+			Assert.assertNotNull(segmentsExperiment);
```

Comment:
Here you are not asserting the update call to AC, so maybe this test should be translated into a unit test or not tested at all.